### PR TITLE
refactor(services): rewrite variables view at service levels

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

> Link to the JIRA ticket

Put description here

> Screenshot of the feature

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [ ] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Make it work, make it right, make it fast

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] my-env - fix: add variables view at project environment levels - 2024-04-19T10:11:58Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/e03b8469-b703-447e-96a4-8ca986873aa5/services/general)

 Another comment will be posted when deployments are finished

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] my-env - fix: add variables view at project environment levels - 2024-04-19T10:11:58Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/e03b8469-b703-447e-96a4-8ca986873aa5/logs)

File: libs/domains/variables/feature/src/lib/variable-list/variable-list.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+const variableListProps: VariableListProps = {
+  parentId: '037c9e87-e098-4970-8b1f-9a5ffe9e4b89',
+  currentScope: 'APPLICATION',
+  onCreateVariable: jest.fn(),
+  onEditVariable: jest.fn(),
+  onDeleteVariable: jest.fn(),
+}
+
+describe('ServiceList', () => {
```
Comment (RemiBonnet) [inline libs/domains/variables/feature/src/lib/variable-list/variable-list.spec.tsx:141]:
```suggestion
describe('VariableList', () => {
```

File: libs/domains/variables/feature/src/lib/variable-list/variable-list.tsx [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+        header: `${variables.length} ${pluralize(variables.length, 'variable')}`,
+        enableColumnFilter: false,
+        size: 40,
+        cell: (info) => {
+          const variable = info.row.original
+          return (
+            <>
+              {variable.owned_by === ExternalServiceEnum.DOPPLER && (
+                <span
+                  data-testid="doppler-tag"
```
Comment (RemiBonnet) [inline libs/domains/variables/feature/src/lib/variable-list/variable-list.tsx:92]:
```suggestion
                  data-testid="doppler-tag"
```
useless ?

File: libs/domains/variables/feature/src/lib/variable-list/variable-list.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    await userEvent.click(screen.getByRole('button', { name: /value/i }))
+    await userEvent.click(screen.getByRole('menuitem', { name: /private/i }))
+    const rows = screen.getAllByRole('row')
+    expect(rows).toHaveLength(2)
+  })
+  it('should have actions', async () => {
+    const { userEvent } = renderWithProviders(<VariableList {...variableListProps} />)
+    const actions = screen.getAllByRole('button', { name: /actions/i })
+    await userEvent.click(actions[0])
+    //
```
Comment (RemiBonnet) [inline libs/domains/variables/feature/src/lib/variable-list/variable-list.spec.tsx:170]:
```suggestion
    
```

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks Camille !

- We have a jump when we use "show all" for values : https://www.loom.com/share/6b0d0aed18a4434bb54c1ccd4aa1999a can we have fixed column ?

- We have null value on the `service link` column
Screenshot 2024-04-23 at 15 40 12

- The mix between uppercase and capitalize on the select filter it not pretty, i know we have the same thing on the other React Table, we can't fix that ?
Screenshot 2024-04-23 at 15 48 10

File: libs/domains/variables/feature/src/lib/variable-list/variable-list.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/domains/variables/feature/src/lib/variable-list/variable-list.tsx:93]: It exists in the old version, I just kept it to be iso

File: libs/domains/variables/feature/src/lib/variable-list/variable-list.spec.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/domains/variables/feature/src/lib/variable-list/variable-list.spec.tsx]: https://github.com/Qovery/console/pull/1296/commits/ea066d714beaa478208bdb58129d2a7e814379ed

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
All look good, except it with the button action and the right padding

Screenshot 2024-04-25 at 14 23 13

**Comment by ctjhoa** [comment]
> All look good, except it with the button action and the right padding
> 
> Screenshot 2024-04-25 at 14 23 13

Fixed 
2024-04-25-145721_955x746_scrot

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**